File: MapboxGLAndroidSDK/src/test/java/com/mapbox/mapboxsdk/location/engine/LocationEngineRequestTest.java
package com.mapbox.mapboxsdk.location.engine;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.junit.MockitoJUnitRunner;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(MockitoJUnitRunner.class)
public class LocationEngineRequestTest {

  @Test
  public void checkDefaultValues() {
    LocationEngineRequest request = new LocationEngineRequest.Builder(1000L).build();

    assertThat(request.getInterval()).isEqualTo(1000L);
    assertThat(request.getDisplacement()).isEqualTo(0.0f);
    assertThat(request.getMaxWaitTime()).isEqualTo(0L);
    assertThat(request.getFastestInterval()).isEqualTo(0L);
  }

  @Test
  public void checkBuilder() {
    LocationEngineRequest request = new LocationEngineRequest.Builder(2000L)
            .setDisplacement(100.0f)
            .setMaxWaitTime(5000L)
            .setFastestInterval(500L)
            .build();

    assertThat(request.getInterval()).isEqualTo(2000L);
    assertThat(request.getDisplacement()).isEqualTo(100.0f);
    assertThat(request.getMaxWaitTime()).isEqualTo(5000L);
    assertThat(request.getFastestInterval()).isEqualTo(500L);
  }

  @Test
  public void checkRequestEqual() {
    LocationEngineRequest request = new LocationEngineRequest.Builder(2000L).build();
    LocationEngineRequest otherRequest = new LocationEngineRequest.Builder(2000L).build();

    assertThat(request).isEqualTo(otherRequest);
    assertThat(request.hashCode()).isEqualTo(otherRequest.hashCode());
  }

  @Test
  public void checkRequestsNotEqual() {
    LocationEngineRequest request = new LocationEngineRequest.Builder(2000L).build();
    LocationEngineRequest otherRequest = new LocationEngineRequest.Builder(2000L)
            .setFastestInterval(500L)
            .build();

    assertThat(request).isNotEqualTo(otherRequest);
  }

  @Test
  public void checkRequestsEqualHashCode() {
    LocationEngineRequest request = new LocationEngineRequest.Builder(2000L).build();
    LocationEngineRequest otherRequest = new LocationEngineRequest.Builder(2000L).build();

    assertThat(request.hashCode()).isEqualTo(otherRequest.hashCode());
  }

  @Test
  public void checkRequestsNonEqualHashCode() {
    LocationEngineRequest request = new LocationEngineRequest.Builder(2000L).build();
    LocationEngineRequest otherRequest = new LocationEngineRequest.Builder(2000L)
            .setFastestInterval(500L)
            .build();

    assertThat(request.hashCode()).isNotEqualTo(otherRequest.hashCode());
  }
}

File: platform/android/MapboxGLAndroidSDK/src/test/java/org/maplibre/android/location/engine/LocationEngineRequestTest.kt
package org.maplibre.android.location.engine

import org.assertj.core.api.Assertions
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.junit.MockitoJUnitRunner

@RunWith(MockitoJUnitRunner::class)
class LocationEngineRequestTest {
    @Test
    fun checkDefaultValues() {
        val request = LocationEngineRequest.Builder(1000L).build()
        Assertions.assertThat(request.interval).isEqualTo(1000L)
        Assertions.assertThat(request.displacement).isEqualTo(0.0f)
        Assertions.assertThat(request.maxWaitTime).isEqualTo(0L)
        Assertions.assertThat(request.fastestInterval).isEqualTo(0L)
    }

    @Test
    fun checkBuilder() {
        val request = LocationEngineRequest.Builder(2000L)
                .setDisplacement(100.0f)
                .setMaxWaitTime(5000L)
                .setFastestInterval(500L)
                .build()
        Assertions.assertThat(request.interval).isEqualTo(2000L)
        Assertions.assertThat(request.displacement).isEqualTo(100.0f)
        Assertions.assertThat(request.maxWaitTime).isEqualTo(5000L)
        Assertions.assertThat(request.fastestInterval).isEqualTo(500L)
    }

    @Test
    fun checkRequestEqual() {
        val request = LocationEngineRequest.Builder(2000L).build()
        val otherRequest = LocationEngineRequest.Builder(2000L).build()
        Assertions.assertThat(request).isEqualTo(otherRequest)
        Assertions.assertThat(request.hashCode()).isEqualTo(otherRequest.hashCode())
    }

    @Test
    fun checkRequestsNotEqual() {
        val request = LocationEngineRequest.Builder(2000L).build()
        val otherRequest = LocationEngineRequest.Builder(2000L)
                .setFastestInterval(500L)
                .build()
        Assertions.assertThat(request).isNotEqualTo(otherRequest)
    }

    @Test
    fun checkRequestsEqualHashCode() {
        val request = LocationEngineRequest.Builder(2000L).build()
        val otherRequest = LocationEngineRequest.Builder(2000L).build()
        Assertions.assertThat(request.hashCode()).isEqualTo(otherRequest.hashCode())
    }

    @Test
    fun checkRequestsNonEqualHashCode() {
        val request = LocationEngineRequest.Builder(2000L).build()
        val otherRequest = LocationEngineRequest.Builder(2000L)
                .setFastestInterval(500L)
                .build()
        Assertions.assertThat(request.hashCode()).isNotEqualTo(otherRequest.hashCode())
    }
}
